### ISSUE TITLE
`git up` Fetching origin interrupted emits backtrace

### ISSUE THREAD

**Issue by liijunwei** [msiemens/PyGitUp#112]
First , thank you for developing this tool, pretty useful~ I used to implement bash script for updating branches myself, but this tool does exactly what I want, what a relief 🙏 

## The problem I met

Actually, if I run `git up` and use ctrl+c to intertupt within 0.5s, 1.5s（before/after "Fetching origin" shown）, I got different backtraces

- within 0.5s(might not precise): due to package not fully imported
    - this case is acceptable for me, as I won't do this a lot
- within 1.5s: due to `git fetch` interrupted but `KeyboardInterrupt` exception was not captured
    - I'd like to hide these backtraces for better experience

```bash
lijunwei@bxzy:PyGitUp(master)$ git up
Fetching origin
^CTraceback (most recent call last):
  File "/usr/local/bin/git-up", line 8, in <module>
    sys.exit(run())
  File "/usr/local/Cellar/pygitup/2.1.0_1/libexec/lib/python3.10/site-packages/PyGitUp/gitup.py", line 572, in run
    gitup.run()
  File "/usr/local/Cellar/pygitup/2.1.0_1/libexec/lib/python3.10/site-packages/PyGitUp/gitup.py", line 186, in run
    self.fetch()
  File "/usr/local/Cellar/pygitup/2.1.0_1/libexec/lib/python3.10/site-packages/PyGitUp/gitup.py", line 324, in fetch
    self.git.fetch(*fetch_args, **fetch_kwargs)
  File "/usr/local/Cellar/pygitup/2.1.0_1/libexec/lib/python3.10/site-packages/PyGitUp/git_wrapper.py", line 191, in fetch
    output = cmd.stdout.read(1)
KeyboardInterrupt
```

## Proposal and disscusstion

I'v prepared a PR to capture this `KeyboardInterrupt` exception, and currently working on test case

I'd like to know how do you like the idea and the code change

pls take a look when you have a moment, comments and disscussion are welcome

**Comment by liijunwei** [comment]
currently
![image](https://user-images.githubusercontent.com/48843657/180585391-b2907edb-dfc3-4889-9791-ec5e0b8cf418.png)
expected
![image](https://user-images.githubusercontent.com/48843657/180585400-8a9fa413-8a47-45dc-98e6-3270780c5ba8.png)

**Comment by liijunwei** [comment]
[x] PR
[x] test case

**Comment by msiemens** [comment]
Closed by your PR in #113 🙂 I agree that returning an exit code and not throwing an exception feels nicer for users!
